lib/matplotlib/axes/_axes.py

```diff
@@ -6292,8 +6292,8 @@ def _normalize_input(inp, ename='input'):
                         facecolor=c,
                         margins=margins))
             else:
-                for x, y, c in reversed(list(zip(xvals, yvals, color))):
-                    split = 2 * len(bins)
+                split = 2 * len(bins)
+                for x, y, c in reversed(list(zip(xvals, yvals, reversed(color)))):
```

Comment:
Oh, you're right. That's interesting; I guess it makes sense since `zip` doesn't supply `__len__`, but it's odd that something like `dict.keys()` won't work with it either.